src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -824,8 +824,35 @@ private static void requestScannedSchematic(@NotNull final StructureName structu
                 final InputStream stream = Structure.getStream(structureName.toString());
                 if (stream != null)
                 {
-                    Log.getLogger().info("BuilderTool: sending schematic " + structureName + "(md5:" + md5 + ") to the server");
-                    MineColonies.getNetwork().sendToServer(new SchematicSaveMessage(Structure.getStreamAsByteArray(stream)));
+                    final UUID id = UUID.randomUUID();
+                    final byte[] structureAsByteArray = Structure.getStreamAsByteArray(stream);
+
+                    if(structureAsByteArray.length <= MAX_MESSAGE_SIZE)
+                    {
+                        MineColonies.getNetwork().sendToServer(new SchematicSaveMessage(structureAsByteArray, id, 1, 1));
+                    }
+                    else
+                    {
+                        final int pieces = structureAsByteArray.length / MAX_MESSAGE_SIZE;
+
+                        Log.getLogger().info("BuilderTool: sending: " + pieces + " pieces with the schematic " + structureName + "(md5:" + md5 + ") to the server");
+                        for (int i = 1; i <= pieces; i++)
+                        {
+                            final int start = (i - 1) * MAX_MESSAGE_SIZE;
+                            final int size;
+                            if (i == pieces)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)